spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/outbound/PubSubMessageHandler.java

```diff
@@ -88,33 +78,10 @@ protected void handleMessageInternal(Message<?> message) throws Exception {
 
 		ListenableFuture<String> pubsubFuture;
 
-		if (payload instanceof PubsubMessage) {
-			pubsubFuture = this.pubSubTemplate.publish(topic, (PubsubMessage) payload);
-		}
-		else {
-			ByteString pubsubPayload;
-
-			if (this.messageConverter != null) {
-				pubsubPayload = ByteString.copyFrom(
-						(byte[]) this.messageConverter.fromMessage(message, byte[].class));
-			}
-			else if (payload instanceof byte[]) {
-				pubsubPayload = ByteString.copyFrom((byte[]) payload);
-			}
-			else if (payload instanceof ByteString) {
-				pubsubPayload = (ByteString) payload;
-			}
-			else {
-				throw new MessageConversionException("Unable to convert payload of type "
-						+ payload.getClass().getName() + " to byte[] for sending to Pub/Sub."
-						+ " Try specifying a message converter.");
-			}
+		Map<String, String> headers = new HashMap<>();
+		this.headerMapper.fromHeaders(message.getHeaders(), headers);
 
-			Map<String, String> headers = new HashMap<>();
-			this.headerMapper.fromHeaders(message.getHeaders(), headers);
-
-			pubsubFuture = this.pubSubTemplate.publish(topic, pubsubPayload, headers);
-		}
+		pubsubFuture = this.pubSubTemplate.publish(topic, payload, headers);
```

Comment:
This side is fine with me as far as end-user can supply any custom `PubSubMessageConverter` into the target `PubSubTemplate`.